lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js

```diff
@@ -66,49 +66,37 @@ module.exports = {
 
           http.integration = this.getIntegration(http, functionName);
 
-          if (http.integration === 'AWS') {
-            if (http.request) {
-              http.request = this.getRequest(http);
-
-              if (http.request.parameters) {
-                http.request.parameters = this.getRequestParameters(http.request);
-              }
-            } else {
-              http.request = {};
-            }
+          if ((http.integration === 'HTTP' || http.integration === 'HTTP_PROXY') &&
+            (!http.request || !http.request.uri)) {
+            const errorMessage = [
+              `You need to set the request uri when using the ${http.integration} integration.`,
+            ];
+            throw new this.serverless.classes.Error(errorMessage);
+          }
 
+          if (http.integration === 'AWS' || http.integration === 'HTTP') {
+            http.request = this.getRequest(http);
             http.request.passThrough = this.getRequestPassThrough(http);
-
-            if (http.response) {
-              http.response = this.getResponse(http);
-            } else {
-              http.response = {};
-            }
-
-            if (http.response.statusCodes) {
-              http.response.statusCodes = _.assign({}, http.response.statusCodes);
-
-              if (!_.some(http.response.statusCodes, code => code.pattern === '')) {
-                http.response.statusCodes['200'] = DEFAULT_STATUS_CODES['200'];
-              }
-            } else {
-              http.response.statusCodes = DEFAULT_STATUS_CODES;
-            }
-          } else if (http.integration === 'AWS_PROXY') {
+            http.response = this.getResponse(http);
+          } else if (http.integration === 'AWS_PROXY' || http.integration === 'HTTP_PROXY') {
             // show a warning when request / response config is used with AWS_PROXY (LAMBDA-PROXY)
             if (http.request) {
               const keys = Object.keys(http.request);
-              if (!(keys.length === 1 && keys[0] === 'parameters')) {
+              const allowedKeys =
+                http.integration === 'AWS_PROXY' ? ['parameters'] : ['parameters', 'uri'];
+
+              if (!_.isEmpty(_.difference(keys, allowedKeys))) {
                 const requestWarningMessage = [
-                  'Warning! You\'re using the LAMBDA-PROXY in combination with a request',
+                  `Warning! You're using the ${http.integration} in combination with a request`,
                   ` configuration in your function "${functionName}". Only the`,
-                  ' \'request.parameters\' configs are available in conjunction with',
-                  ' LAMBDA-PROXY. Serverless will remove this configuration automatically',
+                  _.map(allowedKeys, value => `request.${value}`),
```

Comment:
this winds up showing up as:
`Only therequest.parameters,request.uriconfigs are available in conjunction with HTTP_PROXY.`

can you add a space after `the`, before `configs`, and between the `allowedKeys` (maybe `.join(', ')`)?

Reply:
Sure, I'll look into it a bit later today. Thanks!